### DISCUSSION TITLE
Identify objects using custom id

### DISCUSSION THREAD

**Discussion by james-k-mathew** [getgauge/taiko#2097] [General]
**Describe the bug**
Identify objects using custom id like data-test-id

**To Reproduce**
We are able to identify objects in the following manner, using id, class and href

{class:"btn btn--big btn-primary continue-checkout mobile-hidden"}
{id:"form--checkout_signin_new_user--field--NEW_USER_EMAIL_ADDRESS"}
{href:"?edit_shipping_panel=1"}

We are not able to identify using custom id like data-test-id

{data-test-id:"cart_subtotal"}

For ease of automation this type of id is made available is most places in the application
We are not able to leverage that.
Is there any way to leverage id's like that

**Comment by NivedhaSenthil** [comment]
I am able to successfully locate an element with data attribute,

``` js
 
await goto("https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_global_data");
await highlight(listItem({"data-animal-type":"fish"}));
``` 

Can you please provide a sample where this doesn't work ?

**Comment by james-k-mathew** [comment]
Thanks for the clarification. I tried like the following,

await evaluate(link({"data-test-id":"cart_primarycheckout"}), ele => ele.click());

It is working.
